### PR TITLE
[WIP] build static libraries 

### PR DESCRIPTION
**Pull request by DerThorsten**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by DerThorsten** [comment]
@conda-forge-admin, please rerender

File: recipe/build_folly.sh [hunk 1]
```diff
@@ -50,12 +52,27 @@ if [[ "${target_platform}" == "linux-ppc64le" ]]; then
     CMAKE_ARGS="${CMAKE_ARGS} -DHAVE_VSNPRINTF_ERRORS_EXITCODE=0 -DHAVE_VSNPRINTF_ERRORS_EXITCODE__TRYRUN_OUTPUT=''"
 fi
 
-cmake ${CMAKE_ARGS} -Wno-dev -GNinja -DBUILD_SHARED_LIBS=ON ..
+# Build a shared library for the "folly" package or
+# build a static library for the "folly-static" package.
+if [[ ! -z "${PKG_NAME+x}" && "${PKG_NAME}" == "folly" ]]
+then
```
Comment (xhochy) [inline recipe/build_folly.sh:58]: `PKG_NAME` should always be defined.
```suggestion
if [[ "${PKG_NAME}" == "folly" ]]; then
```

File: recipe/build_folly.sh [hunk 2]
```diff
@@ -50,12 +52,27 @@ if [[ "${target_platform}" == "linux-ppc64le" ]]; then
...
+
+$BUILD_PREFIX/bin/cmake ${CMAKE_ARGS} -Wno-dev -GNinja ..
+
+echo "b) THE PREFIX" $PREFIX 
+echo "b) THE BUILD PREFIX" $BUILD_PREFIX
 
 cat CMakeCache.txt
 
-cmake --build . --parallel
+$BUILD_PREFIX/bin/cmake --build . --parallel
```
Comment (xhochy) [inline recipe/build_folly.sh:74]: This should not be necessary. If a different CMake (not the one in $BUILD_PREFIX) is picked up, this is an indication that something else in your environment is broken.
```suggestion
cmake --build . --parallel 
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,9 +1,10 @@
 {% set version = "2023.08.14.00" %}
 {% set jemalloc_enabled = folly_build_ext is not undefined and folly_build_ext == "jemalloc" %}
 {% set build_ext = "_jemalloc" if jemalloc_enabled else ""  %}
+{% set build_number = 1 %}
```
Comment (xhochy) [inline recipe/meta.yaml:4]: Please keep the build number as plain YAML
```suggestion
```

File: recipe/meta.yaml [hunk 4]
```diff
@@ -13,57 +14,131 @@ source:
     - 0001-Use-CMAKE_SYSTEM_PROCESSOR-instead-of-CMAKE_LIBRARY_.patch
 
 build:
-  number: 0
+  number: {{ build_number }}
```
Comment (xhochy) [inline recipe/meta.yaml:17]:
```suggestion
  number: 1
```

File: recipe/meta.yaml [hunk 5]
```diff
@@ -13,57 +14,131 @@ source:
...
-    - zlib
-    - zstd
-  run:
-    - boost-cpp
-    - jemalloc        # [(not osx) and (folly_build_ext is not undefined and folly_build_ext == "jemalloc")]
-    - jemalloc-local  # [osx and (folly_build_ext is not undefined and folly_build_ext == "jemalloc")]
-    - snappy
+outputs:
+  # Shared library, considered unstable!
+  # See: https://github.com/facebook/folly/#build-notes
```
Comment (xhochy) [inline recipe/meta.yaml:25]: This comment is covered by the run_exports.
```suggestion
```

File: recipe/meta.yaml [hunk 6]
```diff
@@ -13,57 +14,131 @@ source:
...
+  - name: folly
+    script: build_folly.sh   # [unix]
+    script: build_folly.bat  # [win]
+    version: {{ version }}
+    build:
+      number: {{ build_number }}
```
Comment (xhochy) [inline recipe/meta.yaml:31]:
```suggestion
```

File: recipe/meta.yaml [hunk 7]
```diff
@@ -13,57 +14,131 @@ source:
...
+      doc_url: https://github.com/facebook/folly/tree/master/folly/docs
+      dev_url: https://github.com/facebook/folly
+
+  # Static library
+  - name: folly-static
+    script: build_folly.sh   # [unix]
+    script: build_folly.bat  # [win]
+    version: {{ version }}
+    build:
+      number: {{ build_number }}
```
Comment (xhochy) [inline recipe/meta.yaml:86]:
```suggestion
```

File: recipe/meta.yaml [hunk 8]
```diff
@@ -13,57 +14,131 @@ source:
...
+        - libaio  # [linux]
+        - libevent
+        - libsodium
+        - snappy
+        - bzip2
+        - lz4-c
+        - openssl
+        - xz
+        - zlib
+        - zstd
```
Comment (xhochy) [inline recipe/meta.yaml:110]: Please deduplicate the headers, they should only be in the shared build.
```suggestion
        - zstd
        - {{ pin_subpackage('folly', exact=True) }}
```

File: recipe/meta.yaml [hunk 9]
```diff
@@ -13,57 +14,131 @@ source:
...
+      run:
+        - boost-cpp
+        - jemalloc        # [(not osx) and (folly_build_ext is not undefined and folly_build_ext == "jemalloc")]
+        - jemalloc-local  # [osx and (folly_build_ext is not undefined and folly_build_ext == "jemalloc")]
+        - snappy
```
Comment (xhochy) [inline recipe/meta.yaml:115]:
```suggestion
        - snappy
        - {{ pin_subpackage('folly', exact=True) }}
```

File: recipe/meta.yaml [hunk 10]
```diff
@@ -13,57 +14,131 @@ source:
...
+    test:
+      requires:
+        - {{ compiler('cxx') }}
+      commands:
+        - test -f ${PREFIX}/lib/libfolly${SHLIB_EXT}              # [unix]
```
Comment (xhochy) [inline recipe/meta.yaml:65]:
```suggestion
        - test -f ${PREFIX}/lib/libfolly${SHLIB_EXT}              # [unix]
        - test ! -f ${PREFIX}/lib/libfolly.a              # [unix]
```

**Review by xhochy — CHANGES_REQUESTED** [review]
Please have a look at `libprotobuf`. This has some nice further features that ensure that the CMakeConfig is correct in the static case.

**Comment by jjerphan** [comment]
Closing as agreed with @DerThorsten since we do not need the static libraries anymore.

**Pull request closed without merging**